scripts/libraries/selectors.ts

```diff
@@ -9,36 +9,22 @@ export enum FacetCutAction {
 }
 
 export function getSelectors(contract: Contract): string[] {
-  const signatures = Object.keys(contract.interface.functions);
-  const selectors: string[] = signatures.reduce((acc: string[], val: string) => {
-    if (val !== 'init(bytes)') {
-      acc.push(contract.interface.getSighash(val));
-    }
-    return acc;
-  }, []);
+  const selectors: string[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-argument
+  contract.interface.forEachFunction((f) => selectors.push(f.selector));
 
   return selectors;
 }
 
-export function getSelectorsFacet(contract: Contract) {
-  const signatures = Object.keys(contract.interface.functions);
-  // log function names and signatures
-  const selectors = signatures.reduce((acc, val) => {
-    if (val !== 'init(bytes)' && val !== 'supportsInterface(bytes4)') {
-      // @ts-ignore
-      acc.push(contract.interface.getSighash(val));
-    }
-    return acc;
-  }, []);
-
-  return selectors;
+export function getSelectorsFacet(contract: Contract): string[] {
+  return getSelectors(contract);
```

Comment:
Remember that you have the facets from @openzeppelin libs, and the methods of the supportInteface has the same name, which means that for the MockERC721 and the  MockERC1155 you will have an error here because the selector will have the same hash and the cut of the diamond will fails.